hasktorch/src/Data/HList.hs

```diff
@@ -78,14 +94,33 @@ instance HMap f '[] '[] where
 instance (Apply f x y, HMap f xs ys) => HMap f (x ': xs) (y ': ys) where
   hmap f (x :. xs) = apply f x :. hmap f xs
 
+-- | Alternative version of `HMap` with better type inference based on `Apply'`
+class HMap' f xs ys | f xs -> ys where
```

Comment:
I wish. I can't deprecate `HMap` because not for all operations you can define an `Apply'` instance. There's an instance of `Apply` in `Torch.Typed.Tensor` that can't be made into an `Apply'` instance because -- to channel ghc -- the output type is not determined jointly by `f` and the input type...